sql/moz-fx-data-shared-prod/mozilla_vpn_derived/subscription_events_live/view.sql

```diff
@@ -98,39 +98,21 @@ SELECT
   events.event_date,
   events.event_type,
   CASE
-  -- Trial events
   WHEN
     events.event_type IN ("New Trial", "Cancelled Trial")
   THEN
     events.event_type
-  -- "New" events
   WHEN
     events.event_type = "New"
   THEN
-    CASE
-    WHEN
-      all_subscriptions.trial_start IS NOT NULL
-    THEN
-      "Converted Trial"
-    WHEN
-      DATE(all_subscriptions.subscription_start_date) = DATE(all_subscriptions.customer_start_date)
-    THEN
-      "New"
-    ELSE
-      "Resurrected"
-    END
-  -- "Cancelled" events
+    all_subscriptions.subscription_start_reason
   WHEN
-    all_subscriptions.provider = "Apple Store"
+    events.event_type = "Cancelled"
   THEN
-    "Cancelled by IAP"
-  WHEN
-    all_subscriptions.canceled_for_customer_at IS NOT NULL
-    OR all_subscriptions.cancel_at_period_end
-  THEN
-    "Cancelled by Customer"
-  ELSE
-    "Payment Failed"
+    COALESCE(
+      all_subscriptions.ended_reason,
+      IF(provider = "Apple Store", "Cancelled by IAP", "Payment Failed")
+    )
```

Comment:
it seems like ended_reason already includes the logic for `IF(provider = "Apple Store", "Cancelled by IAP", "Payment Failed")`, why is it being duplicated here?

Reply:
Because if an Apple subscription was cancelled and later resurrected its `all_subscriptions.ended_reason` could be null when we're storing the cancelled event here (a situation made more probable by the 7 day delay for VPN ETLs).

That should become less of an issue in the future once your new Apple subscription periods logic from #3329 is in effect that'll produce separate `all_subscriptions` records for the disjoint subscription periods going forward.